File: 01_Language/01_Functions/go/src/functions/ip2long.go
package functions

import "net"

func Ip2long(ipAddress string) uint32 {
	ip := net.ParseIP(ipAddress)
	if ip == nil {
		return 0
	}
	var result uint32
	ipBytes := ip.To4()
	for _, v := range ipBytes {
		result = (result << 8) | uint32(v & 0xFF)
	}
	return result
}

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Ip2long.java
package com.onevgo.functions;

import java.net.InetAddress;
import java.net.UnknownHostException;

public class Ip2long {
    public static long ip2long(String ipAddress) {
        long result = 0;
        try {
            byte[] address = InetAddress.getByName(ipAddress).getAddress();
            for (int i = 0; i < address.length; i++) {
                result = (result << 8) | (address[i] & 0xFF);
            }
        } catch (UnknownHostException e) {
            e.printStackTrace();
        }
        return result;
    }

    public static void main(String[] args) {
        System.out.println(ip2long(Gethostbyname.gethostbyname("www.onevgo.com")));
        System.out.println(ip2long("193.112.23.137"));
        System.out.println(ip2long("999.999.999.999"));
    }
}
